### PR TITLE
Refactor the html for the WAYF in the base theme

### PR DESCRIPTION
**Pull request by Badlapje**
Refactor wayf html & add new html

### CODE REVIEW

**Comment by Badlapje** [comment]
this should be merged after feature/base-theme has been merged

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<form class="wayf__search" method="post" action="{{ action }}">
+    <input type="hidden" name="ID" value="{{ requestId }}"/>{# todo: what's this for? #}
```
Comment (MKodde) [inline theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig:2]: This is to ensure we do not forget the request id. It is later used when the 'ContinueToIdP' module is executed to find the matching AuthNRequest from session.

See: https://github.com/OpenConext/OpenConext-engineblock/blob/1cc2a1848084c69a976c774cb6c8eafef38f4780/library/EngineBlock/Corto/Module/Service/ContinueToIdp.php#L75-L83

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    <input type="hidden" id="form-idp" name="idp" value=""/>{# todo: what's this for? #}
```
Comment (MKodde) [inline theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig:3]: This hidden form field will be filled with the entityID of the selected IdP. Also in the ContinueToIdp module, this value is used to forward the AuthNRequest to.

See:
https://github.com/OpenConext/OpenConext-engineblock/blob/1cc2a1848084c69a976c774cb6c8eafef38f4780/library/EngineBlock/Corto/Module/Service/ContinueToIdp.php#L67-L72

And: 
https://github.com/OpenConext/OpenConext-engineblock/blob/1cc2a1848084c69a976c774cb6c8eafef38f4780/library/EngineBlock/Corto/Module/Service/ContinueToIdp.php#L98-L107

(bar lines 99 through 105)

**Review by MKodde — COMMENTED** [review]
Nice work! See some feedback below.

For your information. The WAYF is not fully rendered using Symfony. The old Corto modules are used to render the WAYF. See the `SingleSignOn` module, specifically the `_showWayf` method. Here you can figure out what certain parts of the template shoud do. 

The `ContinueToIdp` module is tasked with processing the Wayf. this takes place in the `serve` method.

File: languages/messages.en.php [hunk 3]
```diff
@@ -57,6 +58,21 @@
...
+    'wayf_apu'                  => 'Please try again with some different keywords',
+    'wayf_noscript_warning'     => '<p>For this page to function optimally you need JavaScript turned on.</p><p>Without JavaScript you will not be able to remember previously used accounts.  If you want to use that functionality, please enable JavaScript.</p><p><strong>You can, off course, still log in.<strong></strong></p>',
+    'wayf_no_access_account'    => 'No access with this account',
+    'wayf_delete_account'       => 'Delete from your accounts',
+    'wayf_remaining_idps_title' => 'Add an account from the list below',
+    'wayf_select_account'       => 'Select an account from the list below',
+    'wayf_search_placeholder'   => 'Search...',
+    'wayf_your_accounts'        => 'Your accounts',
+    'wayf_add_account'          => 'Add another account',
+    'wayf_no_access_helpdesk'   => 'If you want, you can request access to this service. We will send the request to the ICT-helpdesk of your institution.',
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply languages/messages.en.php:71]: good one! changed it.

File: languages/messages.en.php [hunk 4]
```diff
@@ -42,6 +42,7 @@
     'go_back'               => '&lt;&lt; Go back',
     'note'                  => 'Note',
     'note_no_script'        => 'Since your browser does not support JavaScript, you must press the button below to proceed.',
+    'logo'        => 'logo',
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply languages/messages.en.php:45]: I did, will be adding them along with the translations per theme

[the remark this answers is not in the document]
Comment (Badlapje) [reply languages/messages.en.php:45]: which should be later today (i have it as a todo for after all PR's are merged

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/idp.html.twig [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<article class="wayf__idp" data-keywords="{{ idp['keywords'] }}">
+    <a href="">  {# {{ idp['link'] }} todo ensure this link is in the connectedIdps array #}
+        <picture class="idp__logo">
+            <source srcset="" media="(min-width: 800px)"/>
+            {# todo get different images in different formats #}
+            {# todo update breakpoint #}
+            {# todo get all institutions to describe their logo for the alt #}
+            {# todo get the idp name as an attribute instead of "displaytitle" #}
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/idp.html.twig:8]: Done!

File: theme/base/templates/modules/Authentication/View/Proxy/wayf_content.html.twig [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+{% include '@theme/Default/Partials/LoginBar.html.twig' with { loginName: 'suite_name'|trans } %}
+<main class="wayf">
+    {% include '@theme/Authentication/View/Proxy/Partials/Shared/header.html.twig' with { h2: false } %}
+    {% block wayf_content %}
+        <div class="wayf__content" id="idp-picker">
+            {# todo: do we still need this? what's it for? #}
+            {#                {% include '@theme/Authentication/View/Proxy/Partials/WAYF/contentHeader.html.twig' with {#}
+            {#                    showForm: rememberChoiceFeature#}
+            {#                } %}#}
+
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply theme/base/templates/modules/Authentication/View/Proxy/wayf_content.html.twig]: ah, this makes a lot more sense now!  💯 

Refactored it to reflect this meaning.

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig (hunk 1 above)
Comment (Badlapje) [reply theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig]: clarified it in the code, thx for the explanation :-)

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig (hunk 2 above)
Comment (Badlapje) [reply theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig]: clarified it in the code, thx for the explanation :-)

File: theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <form class="wayf__search" method="post" action="{{ action }}">
-    <input type="hidden" name="ID" value="{{ requestId }}"/>{# todo: what's this for? #}
-    <input type="hidden" id="form-idp" name="idp" value=""/>{# todo: what's this for? #}
+    <input type="hidden" name="ID" value="{{ requestId }}"/>{# needed for the ContinueToIdp module #}
+    <input type="hidden" id="form-idp" name="idp" value=""/>{# needed for the ContinueToIdp module #}
```
Comment (MKodde) [inline theme/base/templates/modules/Authentication/View/Proxy/Partials/WAYF/search.html.twig:3]: :superhero:

**Review by MKodde — APPROVED** [review]

**Pull request merged**